### DISCUSSION TITLE
Connect with AppleID?

### DISCUSSION THREAD

**Discussion by beetlefrosch** [BenjaminOddou/alfred-plex#8] [Q&A] [answered]
I log in to Plex with my AppleID, how do I do that with your Workflow?

**Comment by BenjaminOddou** [comment]
Hello @beetlefrosch, thank you for getting in touch! In the current version, there is no way to connect a Plex account that uses a third party. The main reason for this limitation is that it is not supported in the tool used to interact with the API, [Python-PlexAPI](https://github.com/pkkid/python-plexapi). I have opened an issue regarding this matter, which you can find  [here](https://github.com/pkkid/python-plexapi/issues/1272).

For now, the only way to connect to the workflow would be to provide your Plex authentication token. To do this, you will need to inspect the requests made by your client (https://clients.plex.tv/) to the Plex server to obtain your Plex authentication token (X-Plex-Token). Please note that connecting via an authentication token is not currently part of the workflow, but I can consider updating it to include this functionality in the future.

Feel free to reach out if you need further information or assistance

**Comment by voidone001** [comment]
I also use a 3rd party tool to log in; I use Google. I would definitely like to see the ability to log in with a token. Is there a way to do this currently, or would we need to wait for an update?

**Comment by BenjaminOddou** [comment]
Hello @voidone001, I can update the workflow to accept a plex token instead of login + password, but as explained, you would need to know your authentication plex token. Would it be a good alternative for you ?

**Reply by voidone001** [reply]
If I remember correctly, you can find your token by inspecting any file within your server - by looking at its XML file? Or do you need to go to the plex claim token URL?

**Reply by BenjaminOddou** [reply]
@voidone001 the token you'll find when inspecting the xml will be the server auth token not necessary the same as the account auth token (e.g., if you're not the owner for it would be different). What you can do is : 

- Go into the plex settings under your account panel
- Inspect a request with https://clients.plex.tv/ as the url and check the X-Plex-Token property

Don't hesitate to come back to me in case of questions.

**Comment by BenjaminOddou** [comment]
I didn't mentioned it before but there is also the possibility to add a Plex password to a Google, Facebook, or Apple account, see this [page](https://support.plex.tv/articles/use-federated-authentication-to-sign-in/).

**Comment by BenjaminOddou** [comment, marked as answer]
@voidone001, @beetlefrosch, the [v2.2.0](https://github.com/BenjaminOddou/alfred-plex/releases/tag/v2.2.0) now allows you to sign in using an authentication token. You can follow these [instructions](https://github.com/BenjaminOddou/alfred-plex#plex-accounts--manage-your-plex-accounts) to get your token from your account. Don't hesitate to come back to me in case of questions

**Reply by voidone001** [reply]
Thank you!
